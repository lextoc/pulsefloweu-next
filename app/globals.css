.divider {
  margin: 3rem 0;
}

.content {
  margin-bottom: 25vh;
}

.cards {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(16rem, 20rem));
  gap: 1.5rem;
}
