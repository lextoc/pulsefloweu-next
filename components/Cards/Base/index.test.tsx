import { expect } from "@jest/globals";
import { render } from "@testing-library/react";
import React from "react";

import Card, { CardProps } from "./index";

describe("Card component", () => {
  const defaultProps: CardProps = {
    header: "Header Content",
    content: "Card Content",
    footer: "Footer Content",
  };

  it("renders header, content, and footer", () => {
    const { getByText } = render(<Card {...defaultProps} />);

    const headerElement = getByText("Header Content");
    const contentElement = getByText("Card Content");
    const footerElement = getByText("Footer Content");

    expect(headerElement).toBeInTheDocument();
    expect(contentElement).toBeInTheDocument();
    expect(footerElement).toBeInTheDocument();
  });

  it("renders only header and content when footer is not provided", () => {
    const { getByText, queryByText } = render(
      <Card {...defaultProps} footer={undefined} />,
    );

    const headerElement = getByText("Header Content");
    const contentElement = getByText("Card Content");
    const footerElement = queryByText("Footer Content");

    expect(headerElement).toBeInTheDocument();
    expect(contentElement).toBeInTheDocument();
    expect(footerElement).toBeNull();
  });

  it("applies moving background styles when hasMovingBackground is true", () => {
    const { container } = render(
      <Card {...defaultProps} hasMovingBackground />,
    );

    const cardElement = container.querySelector(".rootMovingBackground");

    expect(cardElement).toBeInTheDocument();
  });

  it("does not apply moving background styles when hasMovingBackground is false", () => {
    const { container } = render(
      <Card {...defaultProps} hasMovingBackground={false} />,
    );

    const cardElement = container.querySelector(".rootMovingBackground");

    expect(cardElement).toBeNull();
  });
});
