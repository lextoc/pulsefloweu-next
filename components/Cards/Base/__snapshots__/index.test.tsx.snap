// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<div>
  <article
    class="root "
  >
    <header
      class="header"
    >
      Header Content
    </header>
    <div
      class="content"
    >
      Card Content
    </div>
    <footer
      class="footer"
    >
      Footer Content
    </footer>
  </article>
</div>
`;
